sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2656,6 +2656,16 @@ object SQLConf {
       .checkValue(_ > 0, "The difference must be positive.")
       .createWithDefault(4)
 
+  val METADATA_CACHE_TTL = buildConf("spark.sql.metadataCacheTTL")
+      .doc("Time-to-live (TTL) value for the metadata caches: partition file metadata cache and " +
+        "session catalog cache. This configuration only has an effect when this value having " +
+        "a positive value. It also requires setting `hive` to " +
+        s"${StaticSQLConf.CATALOG_IMPLEMENTATION} to be applied to the partition file " +
```

Comment:
More conditions for this option to be enabled? https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileStatusCache.scala#L43-L44

Reply:
Updated in https://github.com/apache/spark/pull/28852/commits/18feeb02b11407b9eea1457139a91cb40ddc40cf